### ISSUE TITLE
temporary directory littering

### ISSUE THREAD

**Issue by benibela** [checkpoint-restore/criu#2351]
**Description**

**Steps to reproduce the issue:**
 When many duḿps failed in #2348, it created a bunch of empty directories

**Describe the results you received:**

```
drwx------ 2 root   root   4096 Feb 12 23:48 .criu.temp-aa-policy.8uhTbO
drwx------ 2 root   root   4096 Feb 12 23:51 .criu.temp-aa-policy.90Vqt9
drwx------ 2 root   root   4096 Feb 12 23:29 .criu.temp-aa-policy.9an3IS
drwx------ 2 root   root   4096 Feb 12 23:51 .criu.temp-aa-policy.CBz1ud
drwx------ 2 root   root   4096 Feb 12 23:39 .criu.temp-aa-policy.eA0Kj6
drwx------ 2 root   root   4096 Feb 12 23:34 .criu.temp-aa-policy.GD04gM
drwx------ 2 root   root   4096 Feb 12 23:38 .criu.temp-aa-policy.ItnOKf
drwx------ 2 root   root   4096 Feb 12 23:57 .criu.temp-aa-policy.NuRSjL
drwx------ 2 root   root   4096 Feb 12 23:57 .criu.temp-aa-policy.nUX8Kq
drwx------ 2 root   root   4096 Feb 12 23:36 .criu.temp-aa-policy.PGi24L
drwx------ 2 root   root   4096 Feb 12 23:35 .criu.temp-aa-policy.QNuwGp
drwx------ 2 root   root   4096 Feb 12 23:28 .criu.temp-aa-policy.RyvBPt
drwx------ 2 root   root   4096 Feb 12 23:41 .criu.temp-aa-policy.TXQpQv
drwx------ 2 root   root   4096 Feb 12 23:41 .criu.temp-aa-policy.URYvi0
drwx------ 2 root   root   4096 Feb 12 23:34 .criu.temp-aa-policy.VJixh8
drwx------ 2 root   root   4096 Feb 12 23:48 .criu.temp-aa-policy.whT8gi
```

**Describe the results you expected:**

delete the directories after aborting

**Comment by AbhinavMir** [comment]
I tried to reproduce this on Damn Small Linux (with `qemu-system-x86_64 -drive  file=dsl-2024.alpha.iso,index=0,media=cdrom -drive file=fake_external_hdd.qcow2,index=1,media=disk -m 2048 -boot d`) and then when I had some local problems, I tried to do this on an alpine linux, but ended up having trouble when building from source while running `make`.

```
In function '__conv_val',
    inlined from 'std_strtoul' at compel/plugins/std/string.c:212:7:
compel/plugins/std/string.c:164:24: error: array subscript 97 is above array bounds of 'const char[37]' [-Werror=array-bounds=]
  164 |                 return &conv_tab[__tolower(c)] - conv_tab;
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
compel/plugins/std/string.c: In function 'std_strtoul':
compel/plugins/std/string.c:10:19: note: while referencing 'conv_tab'
   10 | static const char conv_tab[] = "0123456789abcdefghijklmnopqrstuvwxyz";
      |                   ^~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [/var/tmp/criu/scripts/nmk/scripts/build.mk:209: compel/plugins/std/string.o] Error 1
make: *** [Makefile.compel:51: compel/plugins/std.lib.a] Error 2
```

Thought I'd bring this up. Not sure if this deserves an issue.

**Comment by YarBor** [comment]
From the description in issue #2348, I don't know how to reproduce the issue, but I have some ideas to deal with this behavior, let me try it out.
By the way, how can I reproduce this problem (please give me some hints) to verify my idea?

**Comment by YarBor** [comment]
maybe i get it

**Comment by yogesh1801** [comment]
can i work on this issue?

**Comment by avagin** [comment]
This issue was fixed by c750e62caca4791dafc00f0e7e112f5b696d56fa.
